src/bacnet/basic/object/nc.c

```diff
@@ -433,6 +433,24 @@ bool Notification_Class_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
                 uint32_t device_id;
                 BACNET_DESTINATION *destination;
                 BACNET_RECIPIENT *recipient;
+                uint8_t *ft_hour = &TmpNotify.Recipient_List[idx].FromTime.hour;
+                uint8_t *ft_min = &TmpNotify.Recipient_List[idx].FromTime.min;
+                uint8_t *ft_sec = &TmpNotify.Recipient_List[idx].FromTime.sec;
+                uint8_t *tt_hour = &TmpNotify.Recipient_List[idx].ToTime.hour;
+                uint8_t *tt_min = &TmpNotify.Recipient_List[idx].ToTime.min;
+                uint8_t *tt_sec = &TmpNotify.Recipient_List[idx].ToTime.sec;
+
+                if (*ft_hour > 23 || *ft_min > 59 || *ft_sec > 59) {
+                    wp_data->error_class = ERROR_CLASS_PROPERTY;
+                    wp_data->error_code = ERROR_CODE_VALUE_OUT_OF_RANGE;
+                    return false;
+                }
+
+                if (*tt_hour > 23 || *tt_min > 59 || *tt_sec > 59) {
```

Comment:
if (*ft_hour > 23 || *ft_min > 59 || *ft_sec > 59 || *tt_hour > 23 || *tt_min > 59 || *tt_sec > 59 ) {

}

As far as I can see the conditional clause is the same so it should just be the one if statement...